### PR TITLE
[Debt] Upgrade NPM to 18

### LINKED ISSUE #5746: 🛠️ nodejs version 16 soon to be end-of-life
## 🛠️ Tooling

Version 16 of nodejs will be end-of-life later this year, [2023-09-11](https://nodejs.org/en/blog/announcements/nodejs16-eol/).

## 🕵️ Details

[Release schedule](https://github.com/nodejs/release#release-schedule) for nodejs.

## 🙋‍♀️ Proposed Solution

Update nodejs to Active LTS version 18

https://github.com/GCTC-NTGC/gc-digital-talent/blob/f979370980504682e506d0892ebb11ad966ccab9/infrastructure/maintenance-container/Dockerfile#L10-L12

https://github.com/GCTC-NTGC/gc-digital-talent/blob/0dd1915f7490d39e67fbb7a64467630600c0964d/package.json#L7

https://github.com/GCTC-NTGC/gc-digital-talent/blob/f979370980504682e506d0892ebb11ad966ccab9/.nvmrc#L1

https://github.com/GCTC-NTGC/gc-digital-talent/blob/00ba90302d8da00ca8375b66698affe561cc7905/infrastructure/bin/deploy.sh#L23

## ✅ Acceptance Criteria

- [ ] project builds and runs the same using nodejs v18

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #5746 

## 👋 Introduction

This branch upgrades NPM to the latest LTS version - [18.16.0](https://nodejs.org/download/release/latest-v18.x/).

## 🧪 Testing

1. Rebuild the maintenance container with `docker-compose build maintenance`
2. Rerun the `setup` build task in the maintenance container

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/.env.example [hunk 1]
```diff
@@ -4,7 +4,7 @@ OAUTH_POST_LOGOUT_REDIRECT="http://localhost:8000/logged-out"
 
 # Mock auth endpoint
 OAUTH_LOGOUT_URI="http://localhost:8000/oxauth/endsession"
-# SiC endpoint (will break e2e tests)
+# SiC endpoint ( will break e2e tests )
```
Comment (mnigh) [inline apps/web/.env.example:7]: Is there a reason for a space after the opening parenthesis and before the closing parenthesis?

Comment (petertgiles) [reply apps/web/.env.example:7]: Yeah, I just needed something to trigger full CI.  The change itself has no meaning.

Comment (mnigh) [reply apps/web/.env.example:7]: @petertgiles May I suggest another change instead:

`# Logging - enable by setting the logging level.  Eight severity levels from rfc5424.`
to
`# Logging - enable by setting the logging level. Eight severity levels from rfc5424.`

**Review by mnigh — APPROVED** [review]
Just a small nitpick in the `.env.example` file but otherwise...LGTM 👍

**Pull request merged**